src/cmd/manage_nsfs.js

```diff
@@ -596,6 +597,17 @@ async function delete_account(data) {
     write_stdout_response(ManageCLIResponse.AccountDeleted);
 }
 
+async function verify_delete_account(account_email) {
+    const show_secrets = true;
+    const bucket_config_files = await list_config_files(buckets_dir_path, show_secrets);
```

Comment:
list_config_file iterates the buckets_dir_path and reads each file, how much time does this function take when there are thousands of buckets? we need to optimize it (not necessarily on this PR)

There are 2 options that I could think about - 
1. quick and dirty - add to account owned_buckets property - an array of bucket names that the account owns  
2. thorough and meticulous - start working on the rest API, there we will use the accounts_cache of the endpoint, but this will couple between the CLI and the running endpoint (IMHO this is the road we should take)
@guymguym WDYT?

Reply:
Summarizing what we discussed:
We talked about 4 options (the first two you raised here):

1. **Add to the account `owned_buckets` property** - an array of bucket names that the account owns. We will need to manage the buckets changes in two places (the bucket config and the account config for updating the new property `owned_buckets`), and it doesn't align with the practice "Single Source of Truth".
2. **Cache in the endpoint** - use Rest API, there we will use the `accounts_cache `of the endpoint, decided that it would be a limited solution because we need all the list all buckets, what if the cache size is smaller than all the buckets and the bucket that the account owns would not be saved? to have all the data in the cache, would mean to have a cache size that will always save all the buckets.
3. **Health script** - Not to throw an error at the moment the admin tries to delete the account and to allow it. Then, the admin would get a notice about "stale" buckets (buckets without bucket owner) when the health script would run. Then the admin can delete the config files of the stale buckets. Today, it would not prevent a new account from having "stale buckets". Today the bucket owner identifier is the email, hence if we delete an account with buckets, and create a new account with the same email, then the "stale" buckets will be exposed to it (it is a gap that will be solved one the bucket owner identifier will be a generated unique ID).
4. **Optimize `list_config_files` with concurrency (chosen option)** - we see the issue that we are trying to solve (verify that an account doesn't own buckets before deletion) as a private case of the list (in this case list bucket). Hence we need to optimize the function `list_config_files` to improve its performance.